include/envoy/request_info/string_accessor.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace RequestInfo {
+
+/**
+ * Contains a string in a form which is usable with DynamicMetadata and
+ * allows lazy evaluation if needed.
+ */
+class StringAccessor {
```

Comment:
Not sure I fully grok how this is used from the comments here and the rest of the PR.

Reply:
So the basic idea here is that "StringAccessor" is how user defined headers (working on the PR now) get access to DynamicMetadata.  That may mean I should pull it into that PR, and (?) move its definition over to include/envoy/router.  Or maybe I should just expand the comment here to say "Intended to be a globally known interface for accessing string data on DynamicMetadata".  

What's your preference?

Reply:
As discussed offline, pushing to next PR.